imports/plugins/core/versions/server/util/convert68.js

```diff
@@ -0,0 +1,296 @@
+ *   you have already looked them up. This will save a database query.
+ * @param {Object[]} input.canSellWithoutInventory App settings that indicates whether products
+ * can be sold without inventory tracking enabled.
+ * @return {Promise<Object[]>} Array of responses, in same order as `input.productConfigurations` array.
+ */
+async function getInventoryResults(collections, input) {
+  const { SimpleInventory } = collections;
+  const { productConfigurations, canSellWithoutInventory } = input;
+  const results = [];
+  let remainingProductConfigurations = productConfigurations;
+
+  const productVariantIds = productConfigurations.map(({ productVariantId }) => productVariantId);
+
+  const inventoryDocs = await SimpleInventory
+    .find({
+      "productConfiguration.productVariantId": { $in: productVariantIds }
+    })
+    .limit(productConfigurations.length) // optimize query speed
+    .toArray();
+
+  const inventoryResults = productConfigurations.map((productConfiguration) => {
+    const inventoryDoc = inventoryDocs.find((doc) => isEqual(productConfiguration, doc.productConfiguration));
+    if (!inventoryDoc || !inventoryDoc.isEnabled) {
+      return {
+        inventoryInfo: null,
+        productConfiguration
+      };
+    }
+
+    const { canBackorder, lowInventoryWarningThreshold } = inventoryDoc;
+    let { inventoryInStock, inventoryReserved } = inventoryDoc;
+    inventoryInStock = Math.max(0, inventoryInStock);
+    inventoryReserved = Math.max(0, inventoryReserved);
+    const inventoryAvailableToSell = Math.max(0, inventoryInStock - inventoryReserved);
+    const isLowQuantity = inventoryAvailableToSell <= lowInventoryWarningThreshold;
+
+    return {
+      inventoryInfo: {
+        canBackorder,
+        inventoryAvailableToSell,
+        inventoryInStock,
+        inventoryReserved,
+        isLowQuantity
+      },
+      productConfiguration
+    };
+  });
+
+  // Add only those with inventory info to final results.
+  // Otherwise add to sellableProductConfigurations for next run
+  remainingProductConfigurations = [];
+  for (const inventoryResult of inventoryResults) {
+    if (inventoryResult.inventoryInfo) {
+      inventoryResult.inventoryInfo.isSoldOut = inventoryResult.inventoryInfo.inventoryAvailableToSell === 0;
+      inventoryResult.inventoryInfo.isBackorder = inventoryResult.inventoryInfo.isSoldOut && inventoryResult.inventoryInfo.canBackorder;
+      results.push(inventoryResult);
+    } else {
+      remainingProductConfigurations.push(inventoryResult.productConfiguration);
+    }
+  }
+
+
+  // If no inventory info was found for some of the product configs, such as
+  // if there are no plugins providing inventory info, then use default info
+  // that allows the product to be purchased always.
+  if (remainingProductConfigurations.length > 0) {
+    const inventoryInfo = canSellWithoutInventory ? DEFAULT_SELLABLE_INFO : DEFAULT_SOLD_OUT_INFO;
```

Comment:
Seeing this made me realize that there's probably a bug here. It's an existing bug, but it will be more obvious after your changes. When the `canSellVariantWithoutInventory` setting is changed, we would need to update every catalog product where at least one variant/option does not have inventory. Because if this is `true`, then `isSoldOut` needs to be `true` for such variants and `canBackorder` needs to be `false`.

This will require some thought. It doesn't seem good to need to migrate a massive catalog every time a setting switch is flipped. But I can't think of a solution without a more fundamental shift in the way we're doing this.

Reply:
Created an issue about this here: https://github.com/reactioncommerce/reaction/issues/5333